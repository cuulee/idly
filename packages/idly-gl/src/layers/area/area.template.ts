import { AREA } from '../priorities';

export const areaTemplate: {
  selectable: false;
  layer: {
    id: undefined;
    type: 'line';
    source: undefined;
    layout: {
      'line-join': 'round';
      'line-cap': 'round';
    };
    paint: {
      'line-color': '#fff';
      'line-width': 2;
      'line-opacity': 1;
    };
    filter: undefined;
  };
} = {
  selectable: false,
  layer: {
    id: undefined,
    type: 'line',
    source: undefined,
    layout: {
      'line-join': 'round',
      'line-cap': 'round',
    },
    paint: {
      'line-color': '#fff',
      'line-width': 1,
      'line-opacity': 1,
    },
    filter: undefined,
  },
};

export const areaCasingTemplate: {
  selectable: false;
  layer: {
    id: undefined;
    type: 'line';
    source: undefined;
    layout: {
      'line-join': 'round';
      'line-cap': 'round';
    };
    paint: {
      'line-color': '#fff';
      'line-width': {
        base: 4;
        stops: [[16, 4], [18, 18], [22, 20]];
      };
      'line-opacity': {
        base: 0.2;
        stops: [[16, 0.6], [18, 0.3], [22, 0.4]];
      };
      'line-offset': {
        base: 4;
        stops: [[16, 4], [18, 4], [22, 6]];
      };
      'line-blur': {
        base: 2;
        stops: [[16, 0], [18, 0], [22, 0]];
      };
    };
    filter: undefined;
  };
} = {
  selectable: false,
  layer: {
    id: undefined,
    type: 'line',
    source: undefined,
    layout: {
      'line-join': 'round',
      'line-cap': 'round',
    },
    paint: {
      'line-color': '#fff',
      'line-width': {
        base: 4,
        stops: [[16, 8], [18, 16], [22, 40]],
      },
      'line-opacity': {
        base: 0.2,
        stops: [[16, 0.6], [18, 0.3], [22, 0.4]],
      },
      'line-offset': {
        base: 4,
        stops: [[16, 4], [18, 8], [22, 21]],
      },
    },
    filter: undefined,
  },
};
