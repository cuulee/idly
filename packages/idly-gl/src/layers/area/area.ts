import { PLUGIN_NAME } from '../../constants';
import { areaPaintStyle } from './helper';

export default [
  {
    selectable: false,
    priority: 0.1,
    layer: {
      id: 'AreaLayer',
      type: 'line',
      source: undefined,
      layout: {
        'line-join': 'round',
        'line-cap': 'round'
      },
      paint: {
        'line-color': '#551A8B',
        'line-width': 2,
        'line-opacity': 1
      },
      filter: [
        'all',
        ['==', '$type', 'Polygon'],
        /**
         * @REVISIT buildings or any small really look ugly with that gl offset artifact
         *  going for a fill layer for now.
         */
        ['!=', `${PLUGIN_NAME}--tagsClass`, 'tag-building']
      ]
    }
  },
  {
    selectable: false,
    priority: 0.1,
    layer: {
      id: 'AreaLayerCasing',
      type: 'line',
      source: undefined,
      layout: {
        'line-join': 'round',
        'line-cap': 'round'
      },
      paint: areaPaintStyle,
      filter: [
        'all',
        ['==', '$type', 'Polygon'],
        ['!=', `${PLUGIN_NAME}--tagsClass`, 'tag-building']
      ]
    }
  }
];
