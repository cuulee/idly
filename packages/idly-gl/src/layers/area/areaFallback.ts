import { IDLY_NS } from '../../constants';
import { AREA } from '../priorities';
import { areaCasingTemplate, areaTemplate } from './area.template';

export default [
  {
    selectable: false,
    priority: AREA.MINUS_1,
    layer: {
      id: 'AreaFallbackLayer',
      type: 'line',
      source: undefined,
      layout: areaTemplate.layer.layout,
      paint: areaTemplate.layer.paint,
      filter: [
        'all',
        ['==', '$type', 'Polygon'],
        /**
         * @REVISIT buildings or any small really look ugly with that gl offset artifact
         *  going for a fill layer for now.
         */
        ['!=', `${IDLY_NS}tagsClass`, 'tag-building'],
      ],
    },
  },
  {
    selectable: false,
    priority: AREA.MINUS_2,
    layer: {
      id: 'AreaFallbackLayerCasing',
      type: 'line',
      source: undefined,
      layout: areaCasingTemplate.layer.layout,
      paint: {
        'line-color': 'white',
        'line-width': 1,
      },
      filter: [
        'all',
        ['==', '$type', 'Polygon'],
        ['!=', `${IDLY_NS}tagsClass`, 'tag-building'],
      ],
    },
  },
];
