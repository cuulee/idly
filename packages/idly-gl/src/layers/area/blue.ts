import { fromJS } from 'immutable';

import { areaPaintStyle } from '../../layers/area';
import { PLUGIN_NAME } from '../../constants';
import { addSource } from '../../helper/addSource';

const filter = [
  'all',
  ['==', '$type', 'Polygon'],
  [
    'in',
    `${PLUGIN_NAME}--tagsClassType`,
    'tag-amenity-swimming_pool',
    'tag-leisure-swimming_pool',
    'tag-natural-water',
    'tag-landuse-aquaculture',
    'tag-landuse-basin',
    'tag-landuse-harbour',
    'tag-landuse-reservoir'
  ]
];

const blueLayer = {
  id: 'areaBlueLayer',
  type: 'line',
  source: undefined,
  layout: {
    'line-join': 'round',
    'line-cap': 'round'
  },
  paint: {
    'line-color': '#77d3de',
    'line-width': 2,
    'line-opacity': 1
  },
  filter
};

const blueLayerCasing = {
  id: 'areaBlueLayerCasing',
  type: 'line',
  source: undefined,
  layout: {
    'line-join': 'round',
    'line-cap': 'round'
  },
  paint: { ...areaPaintStyle, 'line-color': '#77d3de' },
  filter
};

export const areaBlueLayer = (sourceName: string) =>
  fromJS({
    selectable: false,
    priority: 1,
    layer: addSource(blueLayer, sourceName)
  });

export const areaBlueLayerCasing = (sourceName: string) =>
  fromJS({
    selectable: false,
    priority: 1,
    layer: addSource(blueLayerCasing, sourceName)
  });
