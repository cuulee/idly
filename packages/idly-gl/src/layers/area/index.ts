import { Layer } from '../types';
import area from './areaFallback';
import blue from './blue';
import building from './building';
import gold from './gold';
import gray from './gray';
import green from './green';
import lightGreen from './lightGreen';
import orange from './orange';
import pink from './pink';
import tan from './tan';
import yellow from './yellow';

const layers: Layer[] = [
  ...area,
  ...blue,
  ...building,
  ...gold,
  ...gray,
  ...green,
  ...lightGreen,
  ...orange,
  ...pink,
  ...tan,
  ...yellow,
];

export default layers;
