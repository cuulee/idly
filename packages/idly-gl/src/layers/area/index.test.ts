import area from './';
describe('area layers snapshot', () => {
  const layerIds = area.map(r => r.layer.id);

  it('check the structure', () => {
    expect(area).toHaveLength(22);
    expect(area.map(r => r.layer.id)).toMatchSnapshot();
  });

  for (const layer of layerIds) {
    it('checks layer', () => {
      expect(area.find(r => r.layer.id === layer)).toMatchSnapshot();
    });
  }
});
