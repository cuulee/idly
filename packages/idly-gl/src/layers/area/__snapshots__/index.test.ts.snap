// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`area layers snapshot check the structure 1`] = `
Array [
  "AreaFallbackLayer",
  "AreaFallbackLayerCasing",
  "areaBlueLayer",
  "areaBlueLayerCasing",
  "areaBuildingLayer",
  "areaBuildingLayerFill",
  "areaGoldLayer",
  "areaGoldLayerCasing",
  "areaGrayLayer",
  "areaGrayLayerCasing",
  "areaGreenLayer",
  "areaGreenLayerCasing",
  "areaLightGreenLayer",
  "areaLightGreenLayerCasing",
  "areaOrangeLayer",
  "areaOrangeLayerCasing",
  "areaPinkLayer",
  "areaPinkLayerCasing",
  "areaTanLayer",
  "areaTanLayerCasing",
  "areaYellowLayer",
  "areaYellowLayerCasing",
]
`;

exports[`area layers snapshot checks layer 1`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "area",
      ],
      Array [
        "!has",
        "@idly-tag-building",
      ],
    ],
    "id": "AreaFallbackLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "white",
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 1.75,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 2`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "area",
      ],
      Array [
        "!has",
        "@idly-tag-building",
      ],
    ],
    "id": "AreaFallbackLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "white",
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 1.5,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 3`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-amenity",
          "tag-amenity-swimming_pool",
        ],
        Array [
          "==",
          "@idly-tag-leisure",
          "tag-amenity-swimming_pool",
        ],
        Array [
          "==",
          "@idly-tag-natural",
          "tag-natural-water",
        ],
        Array [
          "in",
          "@idly-tag-landuse",
          "tag-landuse-aquaculture",
          "tag-landuse-basin",
          "tag-landuse-harbour",
          "tag-landuse-reservoir",
        ],
      ],
    ],
    "id": "areaBlueLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#77d3de",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 4`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-amenity",
          "tag-amenity-swimming_pool",
        ],
        Array [
          "==",
          "@idly-tag-leisure",
          "tag-amenity-swimming_pool",
        ],
        Array [
          "==",
          "@idly-tag-natural",
          "tag-natural-water",
        ],
        Array [
          "in",
          "@idly-tag-landuse",
          "tag-landuse-aquaculture",
          "tag-landuse-basin",
          "tag-landuse-harbour",
          "tag-landuse-reservoir",
        ],
      ],
    ],
    "id": "areaBlueLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#77d3de",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 5`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-amenity",
          "tag-amenity-shelter",
        ],
        Array [
          "has",
          "@idly-tag-building",
        ],
      ],
    ],
    "id": "areaBuildingLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "minzoom": 17,
    "paint": Object {
      "line-color": "#e06e5f",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.25,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 6`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-amenity",
          "tag-amenity-shelter",
        ],
        Array [
          "has",
          "@idly-tag-building",
        ],
      ],
    ],
    "id": "areaBuildingLayerFill",
    "paint": Object {
      "fill-color": "#e06e5f",
      "fill-opacity": 0.2,
    },
    "source": undefined,
    "type": "fill",
  },
  "priority": 2.25,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 7`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "in",
        "@idly-tag-landuse",
        "tag-landuse-residential",
        "tag-landuse-construction",
      ],
    ],
    "id": "areaGoldLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#c4bd19",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 8`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "in",
        "@idly-tag-landuse",
        "tag-landuse-residential",
        "tag-landuse-construction",
      ],
    ],
    "id": "areaGoldLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#c4bd19",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 9`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-amenity",
          "tag-amenity-parking",
        ],
        Array [
          "all",
          Array [
            "==",
            "@idly-tag-leisure",
            "tag-leisure-pitch",
          ],
          Array [
            "in",
            "@idly-tag-sport",
            "tag-sport-basketball",
            "tag-sport-skateboard",
          ],
        ],
        Array [
          "in",
          "@idly-tag-landuse",
          "tag-landuse-railway",
          "tag-landuse-quarry",
        ],
        Array [
          "in",
          "@idly-tag-natural",
          "tag-natural-bare_rock",
          "tag-natural-scree",
          "tag-natural-cave_entrance",
          "tag-natural-glacier",
        ],
      ],
    ],
    "id": "areaGrayLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#bbb",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 10`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-amenity",
          "tag-amenity-parking",
        ],
        Array [
          "all",
          Array [
            "==",
            "@idly-tag-leisure",
            "tag-leisure-pitch",
          ],
          Array [
            "in",
            "@idly-tag-sport",
            "tag-sport-basketball",
            "tag-sport-skateboard",
          ],
        ],
        Array [
          "in",
          "@idly-tag-landuse",
          "tag-landuse-railway",
          "tag-landuse-quarry",
        ],
        Array [
          "in",
          "@idly-tag-natural",
          "tag-natural-bare_rock",
          "tag-natural-scree",
          "tag-natural-cave_entrance",
          "tag-natural-glacier",
        ],
      ],
    ],
    "id": "areaGrayLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#bbb",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 11`] = `
Object {
  "layer": Object {
    "filter": Array [
      "any",
      Array [
        "all",
        Array [
          "has",
          "@idly-tag-natural",
        ],
        Array [
          "!in",
          "@idly-tag-natural",
          "tag-natural-beach",
          "tag-natural-sand",
          "tag-natural-scrub",
          "tag-natural-water",
          "tag-natural-bare_rock",
          "tag-natural-scree",
          "tag-natural-cave_entrance",
          "tag-natural-glacier",
        ],
      ],
      Array [
        "all",
        Array [
          "has",
          "@idly-tag-landuse",
        ],
        Array [
          "!in",
          "@idly-tag-landuse",
          "tag-landuse-aquaculture",
          "tag-landuse-basin",
          "tag-landuse-harbour",
          "tag-landuse-reservoir",
          "tag-landuse-residential",
          "tag-landuse-construction",
          "tag-landuse-railway",
          "tag-landuse-quarry",
          "tag-landuse-cemetery",
          "tag-landuse-orchard",
          "tag-landuse-meadow",
          "tag-landuse-farm",
          "tag-landuse-farmland",
          "tag-landuse-retail",
          "tag-landuse-commercial",
          "tag-landuse-landfill",
          "tag-landuse-military",
          "tag-landuse-industrial",
          "tag-landuse-farmyard",
        ],
      ],
      Array [
        "in",
        "@idly-tag-leisure",
        "tag-leisure-nature_reserve",
        "tag-leisure-park",
      ],
      Array [
        "all",
        Array [
          "==",
          "@idly-tag-leisure",
          "tag-leisure-pitch",
        ],
        Array [
          "!in",
          "@idly-tag-sport",
          "tag-sport-basketball",
          "tag-sport-skateboard",
        ],
      ],
    ],
    "id": "areaGreenLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#8cd05f",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 12`] = `
Object {
  "layer": Object {
    "filter": Array [
      "any",
      Array [
        "all",
        Array [
          "has",
          "@idly-tag-natural",
        ],
        Array [
          "!in",
          "@idly-tag-natural",
          "tag-natural-beach",
          "tag-natural-sand",
          "tag-natural-scrub",
          "tag-natural-water",
          "tag-natural-bare_rock",
          "tag-natural-scree",
          "tag-natural-cave_entrance",
          "tag-natural-glacier",
        ],
      ],
      Array [
        "all",
        Array [
          "has",
          "@idly-tag-landuse",
        ],
        Array [
          "!in",
          "@idly-tag-landuse",
          "tag-landuse-aquaculture",
          "tag-landuse-basin",
          "tag-landuse-harbour",
          "tag-landuse-reservoir",
          "tag-landuse-residential",
          "tag-landuse-construction",
          "tag-landuse-railway",
          "tag-landuse-quarry",
          "tag-landuse-cemetery",
          "tag-landuse-orchard",
          "tag-landuse-meadow",
          "tag-landuse-farm",
          "tag-landuse-farmland",
          "tag-landuse-retail",
          "tag-landuse-commercial",
          "tag-landuse-landfill",
          "tag-landuse-military",
          "tag-landuse-industrial",
          "tag-landuse-farmyard",
        ],
      ],
      Array [
        "in",
        "@idly-tag-leisure",
        "tag-leisure-nature_reserve",
        "tag-leisure-park",
      ],
      Array [
        "all",
        Array [
          "==",
          "@idly-tag-leisure",
          "tag-leisure-pitch",
        ],
        Array [
          "!in",
          "@idly-tag-sport",
          "tag-sport-basketball",
          "tag-sport-skateboard",
        ],
      ],
    ],
    "id": "areaGreenLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#8cd05f",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 13`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "in",
        "@idly-tag-landuse",
        "tag-landuse-cemetery",
        "tag-landuse-orchard",
        "tag-landuse-meadow",
        "tag-landuse-farm",
        "tag-landuse-farmland",
      ],
    ],
    "id": "areaLightGreenLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#c7d3a0",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 14`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "in",
        "@idly-tag-landuse",
        "tag-landuse-cemetery",
        "tag-landuse-orchard",
        "tag-landuse-meadow",
        "tag-landuse-farm",
        "tag-landuse-farmland",
      ],
    ],
    "id": "areaLightGreenLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#c7d3a0",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 15`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "in",
          "@idly-tag-landuse",
          "tag-landuse-retail",
          "tag-landuse-commercial",
          "tag-landuse-landfill",
          "tag-landuse-military",
        ],
        Array [
          "has",
          "@idly-tag-military",
        ],
      ],
    ],
    "id": "areaOrangeLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#d6881a",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 16`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "in",
          "@idly-tag-landuse",
          "tag-landuse-retail",
          "tag-landuse-commercial",
          "tag-landuse-landfill",
          "tag-landuse-military",
        ],
        Array [
          "has",
          "@idly-tag-military",
        ],
      ],
    ],
    "id": "areaOrangeLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#d6881a",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 17`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-landuse",
          "tag-landuse-industrial",
        ],
        Array [
          "==",
          "@idly-tag-power",
          "tag-power-plant",
        ],
      ],
    ],
    "id": "areaPinkLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#e4a4f5",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 18`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-landuse",
          "tag-landuse-industrial",
        ],
        Array [
          "==",
          "@idly-tag-power",
          "tag-power-plant",
        ],
      ],
    ],
    "id": "areaPinkLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#e4a4f5",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 19`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "==",
        "@idly-tag-landuse",
        "tag-landuse-farmyard",
      ],
    ],
    "id": "areaTanLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#f5dcba",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 20`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "==",
        "@idly-tag-landuse",
        "tag-landuse-farmyard",
      ],
    ],
    "id": "areaTanLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#f5dcba",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 21`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-sport",
          "tag-sport-beachvolleyball",
        ],
        Array [
          "in",
          "@idly-tag-natural",
          "tag-natural-beach",
          "tag-natural-sand",
          "tag-natural-scrub",
        ],
        Array [
          "in",
          "@idly-tag-amenity",
          "tag-amenity-childcare",
          "tag-amenity-kindergarten",
          "tag-amenity-school",
          "tag-amenity-college",
          "tag-amenity-university",
        ],
      ],
    ],
    "id": "areaYellowLayer",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#ffff94",
      "line-opacity": 1,
      "line-width": 1,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;

exports[`area layers snapshot checks layer 22`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "$type",
        "Polygon",
      ],
      Array [
        "any",
        Array [
          "==",
          "@idly-tag-sport",
          "tag-sport-beachvolleyball",
        ],
        Array [
          "in",
          "@idly-tag-natural",
          "tag-natural-beach",
          "tag-natural-sand",
          "tag-natural-scrub",
        ],
        Array [
          "in",
          "@idly-tag-amenity",
          "tag-amenity-childcare",
          "tag-amenity-kindergarten",
          "tag-amenity-school",
          "tag-amenity-college",
          "tag-amenity-university",
        ],
      ],
    ],
    "id": "areaYellowLayerCasing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#ffff94",
      "line-offset": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            4,
          ],
          Array [
            18,
            8,
          ],
          Array [
            22,
            21,
          ],
        ],
      },
      "line-opacity": Object {
        "base": 0.2,
        "stops": Array [
          Array [
            16,
            0.6,
          ],
          Array [
            18,
            0.3,
          ],
          Array [
            22,
            0.4,
          ],
        ],
      },
      "line-width": Object {
        "base": 4,
        "stops": Array [
          Array [
            16,
            8,
          ],
          Array [
            18,
            16,
          ],
          Array [
            22,
            40,
          ],
        ],
      },
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2,
  "selectable": false,
}
`;
